dttools/src/mq.c

```diff
@@ -0,0 +1,716 @@
+/*
+Copyright (C) 2020- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <poll.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+
+#include "mq.h"
+#include "buffer.h"
+#include "list.h"
+#include "itable.h"
+#include "set.h"
+#include "link.h"
+#include "xxmalloc.h"
+#include "debug.h"
+#include "jx_print.h"
+#include "jx_parse.h"
+
+
+#define HDR_SIZE (sizeof(struct mq_msg) - offsetof(struct mq_msg, magic))
+#define HDR_MAGIC "MQmsg"
+
+enum mq_msg_type {
```

Comment:
Is it necessary to make mq aware of the payload type?
I'm wondering if it is sufficient to just move opaque buffers back and forth, and let the caller interpret it.